VKAPI/Handlers/Wall.php

```diff
@@ -428,6 +440,11 @@ function post(string $owner_id, string $message = "", int $from_group = 0, int $
             $post->setContent($message);
             $post->setFlags($flags);
             $post->setApi_Source_Name($this->getPlatform());
+
+            if(!is_null($copyright) && !empty($copyright) && $copyright != "" && preg_match("/^(http:\/\/|https:\/\/)*[а-яА-ЯёЁa-z0-9\-_]+(\.[а-яА-ЯёЁa-z0-9\-_]+)+(\/\S*)*$/iu", $copyright) && iconv_strlen($copyright) < 50) {
```

Comment:
Почему не `filter_var($copyright, FILTER_VALIDATE_URL)`?
```suggestion
            if(!is_null($copyright) && !empty($copyright) && $copyright != "" && filter_var($copyright, FILTER_VALIDATE_URL) && iconv_strlen($copyright) < 50) {
```

Reply:
С кириллическими доменами не работает

Reply:
🙃
![image](https://github.com/openvk/openvk/assets/93197434/2c4753c1-cef0-414b-b080-d6a511e5ddf8)

Reply:
... А ChatGPT точно стоит верить в данном случае?

Просто эта информация, вроде как, противоречит [докам](https://www.php.net/manual/en/filter.filters.validate.php)

> Note that the function will only find ASCII URLs to be valid; internationalized domain names (containing non-ASCII characters) will fail.

И, та же строка, [но на русском](https://www.php.net/manual/ru/filter.filters.validate.php)

> Обратите внимание, что эта функция считает корректными только URL, состоящие из символов ASCII; Интернациональные доменные имена не пройдут проверку. 

Также, можно же проверить самим. PHP я скачивать не буду, но есть же [OnlinePHP.io](https://onlinephp.io/) :3

Ставим версию, на которой работает оффициальный инстанс (PHP 7.4.26), ну и по приколу другие

![Результаты тестов](https://sxcu.net/6dK5zhje7.png)

Также, FILTER_VALIDATE_URL использовался всего раз, [в приложениях](https://github.com/openvk/openvk/blob/245f8690c681f6327b137ce4423d043ebe8cbe29/Web/Presenters/AppsPresenter.php#L71). И, тут уже не ручаюсь, может я и сделал что-то не так, но вот так это выглядит:

![OVK](https://sxcu.net/6dK9Fhnqd.png)

Так что, @n1rwana, нет, ИИ ещё не захватили мир

Reply:
Один вопрос... А что делать в итоге?
Может вывести в отдельную переменную/функцию, раз FILTER_VALIDATE_URL использовался и в приложениях?